instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/utils/coroutines/TestUtils.kt

```diff
@@ -108,6 +119,27 @@ suspend fun MapboxNavigation.waitForAlternativeRoute() {
     waitForRoutesUpdate(RoutesExtra.ROUTES_UPDATE_REASON_ALTERNATIVE)
 }
 
+@OptIn(ExperimentalPreviewMapboxNavigationAPI::class, ExperimentalCoroutinesApi::class)
+fun MapboxNavigation.collectMetadata(maxItems: Int): Flow<DeveloperMetadata> {
+    val navigation = this
+    return callbackFlow {
+        val observer = object : DeveloperMetadataObserver {
+            override fun onDeveloperMetadataChanged(metadata: DeveloperMetadata) {
+                Log.i("[DeveloperMetadataObserver]", "onDeveloperMetadataChanged: $metadata")
+                trySend(metadata)
```

Comment:
as you don't handle result
```suggestion
                send(metadata)
```

Reply:
I can't do that, send is a suspend function, I'd need a coroutine started here.
It's doe like this everywhere else. BTW, I think that I've placed this method in the wrong file.